src/GafferScene/Wireframe.cpp

```diff
@@ -0,0 +1,256 @@
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#include "GafferScene/Wireframe.h"
+
+#include "Gaffer/StringPlug.h"
+
+#include "IECoreScene/MeshPrimitive.h"
+#include "IECoreScene/CurvesPrimitive.h"
+
+#include "IECore/DataAlgo.h"
+
+#include "boost/functional/hash.hpp"
+
+#include <unordered_set>
+
+using namespace std;
+using namespace Imath;
+using namespace IECore;
+using namespace IECoreScene;
+using namespace Gaffer;
+using namespace GafferScene;
+
+//////////////////////////////////////////////////////////////////////////
+// Internal utilities
+//////////////////////////////////////////////////////////////////////////
+
+namespace
+{
+
+struct MakeWireframe
+{
+
+	CurvesPrimitivePtr operator() ( const V2fVectorData *positions, const vector<int> &verticesPerFace, const vector<int> &vertexIds )
+	{
+		return makeWireframe( positions->baseReadable(), 2, verticesPerFace, vertexIds );
+	}
+
+	CurvesPrimitivePtr operator() ( const V3fVectorData *positions, const vector<int> &verticesPerFace, const vector<int> &vertexIds )
+	{
+		return makeWireframe( positions->baseReadable(), 3, verticesPerFace, vertexIds );
+	}
+
+	CurvesPrimitivePtr operator() ( const Data *positions, const vector<int> &verticesPerFace, const vector<int> &vertexIds )
+	{
+		throw IECore::Exception( boost::str(
+			boost::format( "Position has unsupported type \"%1%\"" ) % positions->typeName()
+		) );
+	}
+
+	private :
+
+		V3f position( const float *positions, int dimensions, int index )
+		{
+			V3f result( 0.0f );
+			for( int i = 0; i < dimensions; ++i )
+			{
+				result[i] = positions[index*dimensions+i];
+			}
+			return result;
+		}
+
+		CurvesPrimitivePtr makeWireframe( const float *positions, int dimensions, const vector<int> &verticesPerFace, const vector<int> &vertexIds )
+		{
+			IECore::V3fVectorDataPtr pData = new V3fVectorData;
+			pData->setInterpretation( GeometricData::Point );
+			vector<V3f> &p = pData->writable();
+
+			using Edge = std::pair<int, int>;
+			using EdgeSet = unordered_set<Edge, boost::hash<Edge>>;
+			EdgeSet edgesVisited;
+
+			int vertexIdsIndex = 0;
+			for( int numVertices : verticesPerFace )
+			{
+				for( int i = 0; i < numVertices; ++i )
+				{
+					int index0 = vertexIds[vertexIdsIndex + i];
+					int index1 = vertexIds[vertexIdsIndex + (i + 1) % numVertices];
+					Edge edge( min( index0, index1 ), max( index0, index1 ) );
+					if( edgesVisited.insert( edge ).second )
```

Comment:
I wonder what the performance of the unordered set is compared to a vector indexed by the min of the two indices and a list of the indices it's connected to.

Reply:
Do you really mean a list? Whatever it is, the nested structure is going to need to support quick membership queries - consider how many edges connect to the vertex at the pole of a highly tessellated sphere.